server/src/main/scala/io/delta/sharing/server/CloudFileSigner.scala

```diff
@@ -188,3 +193,20 @@ object AbfsFileSigner {
       getRelativePath(abfsStore, _))
   }
 }
+
+class GCSFileSigner(
+    name: URI,
+    conf: Configuration,
+    preSignedUrlTimeoutSeconds: Long) extends CloudFileSigner{
+
+  override def sign(path: Path): String = {
+    val absPath = path.toUri
+    val bucketName = absPath.getHost
+    val objectName = absPath.getPath.stripPrefix("/")
```

Comment:
Would be great to have a check `assert(objectName.nonEmpty, s"cannot get object key from $path")` similar to AWS and Azure signer

Reply:
I added it.